### ISSUE TITLE
WebGLRenderer frustum culls using bounding sphere, but sorts using object origin?

### ISSUE THREAD

**Issue by stevesan** [mrdoob/three.js#13857]
https://github.com/mrdoob/three.js/blob/fb46a77f8ef9f82992b5119c834ae72ad1f46578/src/renderers/WebGLRenderer.js#L1318

Wouldn't it be more consistent to also sort by bounding-sphere-center? Then transparent objects don't all need to be centered around their origin. And this would be more consistent with other engines.

**Comment by stevesan** [comment]
To prevent a breaking change, how about adding a flag to THREE.Mesh indicating if it should use bounding-sphere or origin for sorting? boolean THREE.Mesh.sortByBoundingSphere

**Comment by WestLangley** [comment]
>Wouldn't it be more consistent to also sort by bounding-sphere-center?

_Also_? Can you show a code snippet?

> transparent objects don't all need to be centered around their origin. 

Can you please elaborate on that?

**Comment by mrdoob** [comment]
Interesting... We can definitely explore this. 

Any chance you can create a jsfiddle? Should be a matter of creating a couple of transparent cubes and applying a translate transform to the geometry.

Wouldn't consider this a breaking change though?

**Comment by stevesan** [comment]
Sure thing, I'll create a fiddle to demonstrate this.

**Comment by stevesan** [comment]
Alrighty, here's the fiddle: https://jsfiddle.net/wshtoc9o/29/

**Comment by stevesan** [comment]
> Also? Can you show a code snippet?

Oh yeah. I'm referring to the fact that frustum culling is done by bounding sphere, so we're computing it already: https://github.com/mrdoob/three.js/blob/35a26f178c523514673d992da1aece23c1cfca6e/src/math/Frustum.js#L91

**Comment by mrdoob** [comment]
So I guess we mainly have to change [these line](https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js#L1320-L1321) to this?

```js
_vector3.copy( object.geometry.boundingSphere.center )
	.applyMatrix4( object.matrixWorld )
	.applyMatrix4( _projScreenMatrix );
```

**Comment by WestLangley** [comment]
`object.geometry.boundingSphere` can be `null`. For example, if the object is not frustum-culled.

**Comment by gkjohnson** [comment]
Looks like this was fixed in #25974 and #25913
